src/components/image/index.test.tsx

```diff
@@ -75,11 +74,61 @@ describe("Image component", () => {
     });
 
     it("renders an error message if an errorComponent is provided", () => {
+      const emptyThing = SolidFns.createThing();
+      const { asFragment } = render(
+        <Image
+          thing={emptyThing}
+          property="https://example.com/bad-url"
+          errorComponent={({ error }) => (
+            <span id="custom-error-component">{error.toString()}</span>
+          )}
+        />
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
+
+    it("renders default error message if there is an error and no errorComponent is passed", () => {
+      const emptyThing = SolidFns.createThing();
+
+      const { asFragment } = render(
+        <Image thing={emptyThing} property="https://example.com/bad-url" />
+      );
+
+      expect(asFragment()).toMatchSnapshot();
+    });
+
+    it("renders default loading message if thing is undefined and there is no error", () => {
```

Comment:
Reading this it sounds like if there _is_ an error _and_ `thing` is undefined, it should still be showing the error. If it would be showing the loading component instead, no test would currently catch that, so that would be good to add as a test.

Reply:
If there is an error, it should show an error. What changed from before was that we were showing the loading message even when we had finished fetching the Thing and it failed due to an error (thing not found in context or as a prop). Instead we are now displaying the Thing error when there's an error fetching the thing, the value error when a value isn't found for a given property, and the loading message only when we are still effectively fetching the thing (Thing is undefined and there is no error).

Reply:
Ah gotcha, so such a test is not relevant to the scope of this PR :+1: